nominal/core.py

```diff
@@ -863,3 +1000,30 @@ def _verify_csv_path(path: Path | str) -> tuple[Path, FileType]:
     if file_type.extension not in (".csv", ".csv.gz"):
         raise ValueError(f"file {path} must end with '.csv' or '.csv.gz'")
     return path, file_type
+
+
+def _log_timestamp_type_to_conjure(log_timestamp_type: LogTimestampType) -> datasource.TimestampType:
+    if log_timestamp_type == "absolute":
+        return datasource.TimestampType.ABSOLUTE
+    elif log_timestamp_type == "relative":
+        return datasource.TimestampType.RELATIVE
+    raise ValueError(f"timestamp type {log_timestamp_type} must be 'relative' or 'absolute'")
+
+
+def _log_timestamp_type_from_conjure(log_timestamp_type: datasource.TimestampType) -> LogTimestampType:
+    if log_timestamp_type == datasource.TimestampType.ABSOLUTE:
+        return "absolute"
+    elif log_timestamp_type == datasource.TimestampType.RELATIVE:
+        return "relative"
+    raise ValueError(f"unhandled timestamp type {log_timestamp_type}")
+
+
+def _logs_to_conjure(
+    logs: Iterable[Log] | Iterable[tuple[datetime | IntegralNanosecondsUTC, str]],
+) -> Iterable[datasource_logset_api.Log]:
+    for log in logs:
```

Comment:
since logs might be high frequency...may be worth it to map over the iterable depending on the type (rather than checking each element)

Reply:
meh, its too ugly to worry about for now lol